gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,360 @@
+  public static void afterClass() throws ExecutionException, InterruptedException {
+    if (storage != null && !RemoteGcsHelper.forceDelete(storage, BUCKET, 5, TimeUnit.SECONDS)) {
+      if (log.isLoggable(Level.WARNING)) {
+        log.log(Level.WARNING, "Deletion of bucket {0} timed out, bucket is not empty", BUCKET);
+      }
+    }
+  }
+
+  public static void fillFile(Storage storage, String fname, int size) throws IOException {
+    try (WriteChannel writer = storage.writer(BlobInfo.builder(BUCKET, fname).build())) {
+      byte[] bytes = new byte[size];
+      Random rnd = new Random(size);
+      rnd.nextBytes(bytes);
+      writer.write(ByteBuffer.wrap(bytes));
+    }
+  }
+
+  @Test
+  public void testFileExists() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.exists(path)).isTrue();
+  }
+
+  @Test
+  public void testFileSize() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.size(path)).isEqualTo(SML_SIZE);
+  }
+
+  @Test
+  public void testReadByteChannel() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    long size = Files.size(path);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    ByteBuffer buf = ByteBuffer.allocate(1024 * 1024);
+    int timeout = 1024 * 1024;
+    int read = 0;
+    while (chan.isOpen()) {
+      if (timeout-- <= 0) {
+        break;
+      }
+      int rc = chan.read(buf);
+      if (rc < 0) {
+        // EOF
+        break;
+      }
+      buf.clear();
+      assertThat(rc).isGreaterThan(0);
+      read += rc;
+      assertThat(chan.position() == read);
+    }
+    assertThat(read).isEqualTo(size);
+  }
+
+  @Test
+  public void testSeek() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(BIG_FILE);
+    long size = Files.size(path);
+    final ByteBuffer buf1a = ByteBuffer.allocate(100);
+    final ByteBuffer buf1b = ByteBuffer.allocate(100);
+    final ByteBuffer buf2a = ByteBuffer.allocate(100);
+    final ByteBuffer buf2b = ByteBuffer.allocate(100);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    reallyRead(chan, buf1a);
+    long dest = size / 2;
+    chan.position(dest);
+    reallyRead(chan, buf2a);
+    // now go back and read it again
+    // (we do 2 locations because 0 is sometimes a special case).
+    chan.position(0);
+    reallyRead(chan, buf1b);
+    chan.position(dest);
+    reallyRead(chan, buf2b);
+    // if the two spots in the file have the same contents, then this isn't a good file for this
+    // test.
+    assertThat(buf1a.array()).isNotEqualTo(buf2a.array());
```

Comment:
I think a better test would have been against a known content (e.g. this test will not detect a seek that is offset by X).